lib/src/rules/unrelated_type_equality_checks.dart

```diff
@@ -126,11 +129,19 @@ class DerivedClass2 extends ClassBase with Mixin {}
 
 ''';
 
+const String _dartCoreLibraryName = 'dart.core';
+
 bool _hasNonComparableOperands(BinaryExpression node) =>
     !DartTypeUtilities.isNullLiteral(node.leftOperand) &&
     !DartTypeUtilities.isNullLiteral(node.rightOperand) &&
     DartTypeUtilities.unrelatedTypes(
-        node.leftOperand.bestType, node.rightOperand.bestType);
+        node.leftOperand.bestType, node.rightOperand.bestType) &&
+    (node.rightOperand.bestType.name != 'int' ||
+        node.rightOperand.bestType.element?.library?.name !=
+            _dartCoreLibraryName ||
+        !((node.leftOperand.bestType.name == 'Int32' ||
+                node.leftOperand.bestType.name == 'Int64') &&
+            node.leftOperand.bestType.element?.library?.name == 'fixnum'));
```

Comment:
Please extract `static bool _isCoreInt(DartType)` and `static bool _isFixNumInt32/64` helpers to make code clearer.

Reply:
Done.